src/hotspot/share/runtime/globals.hpp

```diff
@@ -2009,6 +2009,11 @@ const int ObjectAlignmentInBytes = 8;
                                                                             \
   product(bool, ProfileExceptionHandlers, true,                             \
           "Profile exception handlers")                                     \
+                                                                            \
+  product(bool, AlwaysRecordEvolDependencies, true, DIAGNOSTIC,             \
```

Comment:
As we record all dependencies not only evol_method ones, should we name it just: `AlwaysRecordDependencies`?

Reply:
That’s not exactly right either.  `RecordAllDependencies` would be more like it.  Because:

  - We might record some dependencies that we know we need, and leave others out.
  - Or, we might record all dependencies, even ones we think we might not need.

(But we will need them all if somebody turns on JFR.)

I like this change.  Having a diagnostic switch means we can do a rough triage if something seems to go wrong with this change, down the road.

Reply:
No, because you want to turn on/off evol_method independently of the other dependencies that the compiler is recording.